code/modules/antagonists/hivemind/hivemind.dm

```diff
@@ -135,11 +139,28 @@
 			assim_objective.find_target_by_role(role = ROLE_HIVE, role_type = 1, invert = 1)
 		assim_objective.update_explanation_text()
 		objectives += assim_objective
+	else
+		var/datum/objective/hivemind/biggest/biggest_objective = new
+		biggest_objective.owner = owner
+		objectives += biggest_objective
+
+	if(prob(85) && istype(SSticker.mode, /datum/game_mode/hivemind) && SSticker.mode.assimilation_obj) //If the mode rolled the versus objective, add a very high chance to get this
+		var/datum/objective/hivemind/assimilate_common/versus_objective = new
+		versus_objective.owner = owner
+		versus_objective.target = SSticker.mode.assimilation_obj.target
+		versus_objective.update_explanation_text()
+		objectives += versus_objective
 	else if(prob(70))
+		var/giveit = TRUE
 		var/datum/objective/assassinate/kill_objective = new
 		kill_objective.owner = owner
 		kill_objective.find_target()
-		objectives += kill_objective
+		for(var/datum/objective/hivemind/assimilate/ass_obj in objectives)
```

Comment:
This could use moving to core objective code, but i guess it can be done in another PR.